dtm0/recovery.c

```diff
@@ -0,0 +1,2153 @@
+IvanA: Agree. I will add it once we are more-or-less confident in use-cases and
+       requirements.
+<<<---
+--->>>
+Max: [* style *] usecases section should go to .h file, along with the rest of
+     the sections in .h in DLD template.
+<<<---
+
+
+   <hr>
+   @section DTM0BR-ovw Overview
+   <i>All specifications must start with an Overview section that
+   briefly describes the document and provides any additional
+   instructions or hints on how to best read the specification.</i>
+
+   The document describes the way how DTM0 service is supposed to restore
+   consistency of the data replicated across a certain kind of Motr-based
+   cluster. The term "basic" implies that consistency is resored only
+   for a limited number of use-cases. In the rest of the document,
+   the term "basic" is omitted for simplicity (there are no other kinds of
+   DTM0-based recovery at the moment).
+
+--->>>
+Max: [* defect *] This is an obvious definition, which has very small value.
+     Please make an overview which would bring as much useful information to the
+     reader as it can in just a few lines.
+IvanA: [fixed] I tried to stick to the point, and avoid any "new" terms in
+       the paragraph.
+<<<---
+
+
+   <hr>
+   @section DTM0BR-def Definitions
+   <i>Mandatory.
+   The DLD shall provide definitions of the terms and concepts
+   introduced by the design, as well as the relevant terms used by the
+   specification but described elsewhere.  References to the
+   M0 Glossary and the component's HLD are permitted and encouraged.
+   Agreed upon terminology should be incorporated in the glossary.</i>
+
+--->>>
+Max: [* question *] Are those all the terms that are used here? If no, please
+     add the rest of the terms. Please also consider adding terms that are being
+     used, but which are not in DTM0 HLD.
+IvanA: [fixed] No, it seems like they are not used. I removed this part.
+       I'll populate the "new terms" first, and if there are any redundant
+       information, I'll add references to the HLD.
+<<<---
+
+   Terms and definitions used in the document:
+   - <b>User service</b> is a Motr service that is capable of replicating
+     of its data ("user data") across the cluster. The document is focused on
+     CAS (see @ref cas-dld) but this term could be applied to any Motr service
+     that supports CRDT[2] and has behavior similar to CAS.
+     Note, this document does not differentiate actual "clients" and "servers".
+     For example, the Motr client (including DIX) is also considered to be a
+     "User service".
+   - <b>DTM0 service</b> is a Motr-service that helps a user service restore
+     consistency of its replicated data.
+   - <b>Recoverable process</b> is Motr process that has exactly one user
+     service and exactly one DTM0 service in it. Note that in UT environment,
+     a single OS process may have several user/DTM0 services, thus it may
+     have multiple recoverable services.
```

Comment:
Fixed.

```
105    - <b>Recoverable process</b> is Motr process that has exactly one user
 106      service and exactly one DTM0 service in it. Note that in UT environment,
 107      a single OS process may have several user/DTM0 services, thus it may
 108      have multiple recoverable processes.
```